zerver/lib/github.py

```diff
@@ -12,7 +14,11 @@
 
 class GithubSession(OutgoingSession):
     def __init__(self, **kwargs: Any) -> None:
-        super().__init__(role="github", timeout=5, **kwargs)
+        github_api_auth_token = settings.GITHUB_API_AUTH_TOKEN
```

Comment:
You should add a commented section to `zproject/prod_settings_template.py` (search for "Previews") which doesn't have any settings, but does describe that this is a secret that can be set, and how to generate the token (e.g. can one use [fine-grained access tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#fine-grained-personal-access-tokens), and is so what rights are needed?).

As written, it's not *required* to get previews, but setting it will allow users to make better use of them -- e.g. describing that it can give access to private repos seems like a good thing to note and will make folks potentially more interested in the feature.  Mentioning what the quotas are with and without a token would also be good.